.github/workflows/release-plugins.yaml

```diff
@@ -0,0 +1,164 @@
+name: Backstage plugin release
+
+on:
+  pull_request:
+    branches:
+      - 'release-*'
+    paths:
+      - 'backstage/plugins/knative-event-mesh-backend/**'
+  push:
+    branches:
+      - 'release-*'
+      # Do not run on push to release-next branch. We don't want to publish the plugin on every push to release-next.
+      # It might be okay to do so, but sobranch tool doesn't work well with release-next branch name.
+      - '!release-next'
+# Ideally, we would want to only publish the plugin on NPM when there's an actual code change in the plugin.
+# However, using the path filters block workflow execution on initial branch creation.
+# So, let's publish on every push to a release branch.
+# see https://github.com/openshift-knative/backstage-plugins/pull/15#issuecomment-2333324099
+#    paths:
+#      - 'backstage/plugins/knative-event-mesh-backend/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          filters: |
+            event_mesh_plugin:
+              - 'backstage/plugins/knative-event-mesh-backend/**'
+
+      - name: Setup Golang
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Checkout openshift-knative/hack
+        uses: actions/checkout@v4
+        with:
+          repository: 'openshift-knative/hack'
+          ref: 'main'
+          path: ./src/github.com/openshift-knative/hack
+
+      - name: Install sobranch
+        run: |
+          go install ./cmd/sobranch
+        working-directory: ./src/github.com/openshift-knative/hack
+
+      - name: Setup release name for release branch push
+        if: github.event_name == 'push' && steps.changes.outputs.event_mesh_plugin == 'true'
```

Comment:
same issue as before, this will not run when branch is created `github.event.created`

Reply:
Also this step doesn't need to be guarded

Reply:
This step needs to be guarded, because it will also run on PRs against `release-next`. `sobranch` tool will fail with `release-next` as the input.